packages/agreement-process/src/model/domain/agreement-validators.ts

```diff
@@ -189,6 +194,30 @@ export const assertActivableState = (agreement: Agreement): void => {
   }
 };
 
+export const assertIsDelegate = (
+  delegation: Delegation,
+  delegateId: TenantId
+): void => {
+  if (delegation.delegateId !== delegateId) {
+    throw operationRestrictedToDelegate(delegateId, delegation.id);
+  }
+};
+
+export const assertDelegationIsActive = (delegation: Delegation): void => {
+  if (delegation.state !== delegationState.active) {
+    throw delegationNotActive(delegation.id);
+  }
+};
+
+export const assertTenantIsRequester = (
+  organizationId: TenantId,
+  tenantId: TenantId
+): void => {
+  if (organizationId !== tenantId) {
+    throw tenantIsNotRequester(organizationId, tenantId);
+  }
+};
+
```

Comment:
Here I am starting to see the scary conflicts with #1175 since they add similar validators and a lot of logic to agreement 🥲 
Even after this is approved, I would wait for Capofila to be merged and bring the changes into `feature/incaricato` before merging this, so that we can solve the conflicts in this PR